logging/src/test/java/org/mule/runtime/test/integration/logging/SingleAppModeLoggerTestCase.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.test.integration.logging;
+
+import static org.mule.tck.probe.PollingProber.probe;
+import static org.mule.test.allure.AllureConstants.ArtifactDeploymentFeature.SingleAppDeploymentStory.SINGLE_APP_DEPLOYMENT;
+import static org.mule.test.allure.AllureConstants.IntegrationTestsFeature.INTEGRATIONS_TESTS;
+import static org.mule.test.allure.AllureConstants.Logging.LOGGING;
+import static org.mule.test.infrastructure.FileContainsInLine.hasLine;
+
+import static java.lang.String.format;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.module.deployment.impl.internal.builder.ApplicationFileBuilder;
+import org.mule.runtime.module.deployment.impl.internal.builder.JarFileBuilder;
+import org.mule.runtime.module.log4j.internal.ApplicationReconfigurableLoggerContextSelector;
+import org.mule.tck.util.CompilerUtils;
+import org.mule.test.infrastructure.deployment.AbstractFakeMuleServerTestCase;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.function.Consumer;
+
+import io.qameta.allure.Feature;
+import io.qameta.allure.Features;
+import io.qameta.allure.Story;
+import org.junit.Rule;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.bridge.SLF4JBridgeHandler;
+
+@Features({@Feature(INTEGRATIONS_TESTS), @Feature(LOGGING)})
+@Story(SINGLE_APP_DEPLOYMENT)
+public class SingleAppModeLoggerTestCase extends AbstractFakeMuleServerTestCase {
+
+  public static final String FIRST_LOG = "My logger is LOG4J";
+  public static final String SECOND_LOG = "My logger is JUL";
+  public static final String THIRD_LOG = "My logger is LOG4J";
+  public static final String FOURTH_LOG = "My logger is JCL";
+  public static final String LOG_MESSAGE = "This log should be present in the app log.";
+  private ApplicationReconfigurableLoggerContextSelector contextSelector = new ApplicationReconfigurableLoggerContextSelector();
+  @Rule
+  public UseMuleLog4jContextFactory muleLogging =
+      new UseMuleLog4jContextFactory(contextSelector);
+
+  @Override
+  public void setUp() throws Exception {
+    super.setUp();
+    SLF4JBridgeHandler.removeHandlersForRootLogger();
+    SLF4JBridgeHandler.install();
+  }
+
+  @Override
+  public void tearDown() throws Exception {
+    super.tearDown();
+    SLF4JBridgeHandler.uninstall();
+  }
+
+  @Test
+  public void log4jLogsInAppSuccessfully() throws Exception {
```

Comment:
make clear the scope of what this is testing. According to our talk in slack, this is only testing the reconfiguration.